src/util/web_worker_transfer.js

```diff
@@ -68,16 +69,18 @@ export function register<T: any>(name: string, klass: Class<T>, options: Registe
 
 register('Object', Object);
 
-Grid.serialize = function serializeGrid(grid: Grid, transferables?: Array<Transferable>): Serialized {
-    const ab = grid.toArrayBuffer();
+type SerializedGrid = { buffer: ArrayBuffer };
+
+Grid.serialize = function serializeGrid(grid: Grid, transferables?: Array<Transferable>): SerializedGrid {
```

Comment:
I'm missing something... what was the JSON savings in changing from serializing as an `ArrayBuffer` to serializing as a `SerializedGrid`?

Reply:
If you serialize just the `ArrayBuffer` which is transferred, any custom properties on it (in this case `$name`) don't survive the transfer. Before the PR we implicitly transferred in a wrapper (`{name: 'Grid', properties: buffer}`), and now we explicitly transfer as `{$name: 'Grid', buffer}`. I'm now throwing an error is `serialize` in case the serialized object is fully transferred just so that we don't trip up on this in future.

Reply:
Ah got it, thanks.